src/Store/VimStoreConnector.re

```diff
@@ -525,15 +525,36 @@ let start =
   let initEffect =
     Isolinear.Effect.create(~name="vim.init", () => {
       Vim.init();
-      let _ = Vim.command("e oni://Welcome");
-      hasInitialized := true;
 
-      let bufferId = Vim.Buffer.getCurrent() |> Vim.Buffer.getId;
-      dispatch(
-        Actions.BufferRenderer(
-          BufferRenderer.RendererAvailable(bufferId, BufferRenderer.Welcome),
-        ),
-      );
+      if (Core.BuildInfo.commitId == Persistence.Global.version()) {
+        let _ = Vim.command("e oni://Welcome");
+        hasInitialized := true;
+
+        let bufferId = Vim.Buffer.getCurrent() |> Vim.Buffer.getId;
+        dispatch(
+          Actions.BufferRenderer(
+            BufferRenderer.RendererAvailable(
+              bufferId,
+              BufferRenderer.Welcome,
+            ),
+          ),
+        );
+      } else {
+        let _ = Vim.command("e oni://UpdateChangelog");
+        hasInitialized := true;
+
+        let bufferId = Vim.Buffer.getCurrent() |> Vim.Buffer.getId;
+        dispatch(
+          Actions.BufferRenderer(
+            BufferRenderer.RendererAvailable(
+              bufferId,
+              BufferRenderer.UpdateChangelog({
+                since: Persistence.Global.version(),
+              }),
+            ),
+          ),
+        );
+      };
```

Comment:
Unfortunately I had to replace the Welcome screen with the Update screen as opening both would have it fo into some kind of infinite `BufferEnter` loop.